File: legacy/arena-backend/src/modules/device/device.controller.ts
import { Body, Controller, Post, UseGuards } from '@nestjs/common';
import {
  ApiBearerAuth,
  ApiOkResponse,
  ApiOperation,
  ApiTags,
} from '@nestjs/swagger';
import { ArenaJwtAuthGuard } from 'src/guards/arena-jwt-auth-guard';
import { SessionGuard } from '../session/session.guard';
import { ApiResultDto } from 'src/dtos/api-result.dto';
import { RegisterDeviceDto } from './dtos/register-device.dto';
import { UnregisterDeviceDto } from './dtos/unregister-device.dto';
import { DeviceService } from './device.service';
import { CurrentUser } from 'src/decorators/current-user.decorator';
import type { CachedUser } from '../session/session.types';

@ApiTags('devices')
@Controller('/api/v1/devices')
@UseGuards(ArenaJwtAuthGuard, SessionGuard)
@ApiBearerAuth()
export class DeviceController {
  constructor(private readonly deviceService: DeviceService) {}

  @Post('register')
  @ApiOperation({ summary: '디바이스 등록', operationId: 'registerDevice' })
  @ApiOkResponse({ type: ApiResultDto })
  async registerDevice(
    @CurrentUser() user: CachedUser,
    @Body() dto: RegisterDeviceDto,
  ): Promise<ApiResultDto> {
    await this.deviceService.registerDevice(user.userId, dto);
    return {
      success: true,
      errorCode: null,
    };
  }

  @Post('unregister')
  @ApiOperation({ summary: '디바이스 해제', operationId: 'unregisterDevice' })
  @ApiOkResponse({ type: ApiResultDto })
  async unregisterDevice(
    @Body() dto: UnregisterDeviceDto,
  ): Promise<ApiResultDto> {
    await this.deviceService.unregisterDevice(dto);
    return {
      success: true,
      errorCode: null,
    };
  }
}

File: legacy/backend/src/main/java/app/sandori/arena/api/domain/device/DeviceController.java
package app.sandori.arena.api.domain.device;

import app.sandori.arena.api.domain.device.dtos.RegisterDeviceRequest;
import app.sandori.arena.api.domain.device.dtos.UnregisterDeviceRequest;
import app.sandori.arena.api.domain.session.CachedUser;
import app.sandori.arena.api.domain.session.RequireSession;
import app.sandori.arena.api.global.dto.ApiResult;
import app.sandori.arena.api.security.CurrentUser;
import io.swagger.v3.oas.annotations.Operation;
import io.swagger.v3.oas.annotations.tags.Tag;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@Tag(name = "devices")
@RestController
@RequestMapping("/api/v1/devices")
@RequiredArgsConstructor
public class DeviceController {

    private final DeviceService deviceService;

    @PostMapping("register")
    @RequireSession
    @Operation(summary = "디바이스 등록", operationId = "registerDevice")
    public ApiResult registerDevice(
            @CurrentUser CachedUser user,
            @RequestBody RegisterDeviceRequest request) {
        deviceService.registerDevice(user.userId(), request);
        return ApiResult.success();
    }

    @PostMapping("unregister")
    @RequireSession
    @Operation(summary = "디바이스 해제", operationId = "unregisterDevice")
    public ApiResult unregisterDevice(@RequestBody UnregisterDeviceRequest request) {
        deviceService.unregisterDevice(request);
        return ApiResult.success();
    }
}
